airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -204,6 +222,136 @@ def test_read_sequential_slices(
+        slice checkpoints resulting in batches of messages that look like:
+        <state message>
+        <record message>
+        ...
+        <record message>
+
+        Using these batches, we then make additional read method calls using the state message and verify the correctness of the
+        messages in the response.
+        """
+        if inputs.skip_comprehensive_incremental_tests:
+            pytest.skip("Skipping new incremental test based on acceptance-test-config.yml")
+            return
+
+        threshold_days = getattr(inputs, "threshold_days") or 0
+        stream_mapping = {stream.stream.name: stream for stream in configured_catalog_for_incremental.streams}
+
+        output = docker_runner.call_read(connector_config, configured_catalog_for_incremental)
+        records_1 = filter_output(output, type_=Type.RECORD)
+        states_1 = filter_output(output, type_=Type.STATE)
+
+        assert states_1, "Should produce at least one state"
+        assert records_1, "Should produce at least one record"
+
+        # If an outbound state message is not in the per-stream format we should skip the test for the connector being tested
+        if not is_per_stream_state(states_1[-1]):
+            pytest.skip("Skipping per-stream state incremental tests because the connector has not been migrated to the new state format")
+            return
+
+        latest_per_stream_by_name = construct_latest_state_from_messages(states_1)
+
+        for record_value, state_value, stream_name in records_with_state(
+            records_1, latest_per_stream_by_name, stream_mapping, cursor_paths
+        ):
+            assert (
+                record_value <= state_value
+            ), f"First incremental sync should produce records younger or equal to cursor value from the state. Stream: {stream_name}"
+
+        # Create partitions made up of one state message followed by any records that come before the next state
+        filtered_messages = [message for message in output if message.type == Type.STATE or message.type == Type.RECORD]
+        right_index = len(filtered_messages)
+        checkpoint_messages = []
+        for index, message in reversed(list(enumerate(filtered_messages))):
+            if message.type == Type.STATE:
+                message_group = (filtered_messages[index], filtered_messages[index + 1 : right_index])
+                checkpoint_messages.insert(0, message_group)
+                right_index = index
+
+        # We sometimes have duplicate identical state messages in a stream which we can filter out to speed things up
+        checkpoint_messages = [message for index, message in enumerate(checkpoint_messages) if message not in checkpoint_messages[:index]]
+
+        # To avoid spamming APIs we only test a fraction of batches (10%) and enforce a minimum of 10 tested
+        min_batches_to_test = 10
```

Comment:
I increased the sample percentage from 5% to 10% just to make the test a little more useful than before